### DISCUSSION TITLE
Typst's variables aren't highlighted properly, is that a Helix's problem, Tree Sitter grammar for Typst's problem, or a misconfiguration?

### DISCUSSION THREAD

**Discussion by tapyu** [helix-editor/helix#10754] [Troubleshooting] [answered]
I am sorry for my ignorance, but I am a bit naive when it comes to terminal text editor configuration, so I am a bit lost here.

AFAIK, the grammar tree is what provides syntax highlighting, not the LSP or the DAP. Helix now has a [built-in grammar tree-sitter for Typst](https://github.com/uben0/tree-sitter-typst/), and it is indeed working.

However, neither variables nor code section block (i.e., `#{}`) are being highlighted

![image](https://github.com/helix-editor/helix/assets/22801918/5de0ed54-6d05-4568-b012-0a60e0c18eb0)

Where the `variable` name shown in the floating box comes from `:tree-sitter-highlight-name`, thus indicating that the grammar tree-sitter is aware that `my_variable` is indeed a variable, but somehow it doesn't get highlighted.

- Is that a Helix's issue?
- Is that a limitation in this grammar tree-sitter?
- Is it possible to configure Helix to make it highlighted?

Thank you beforehand!

**Comment by the-mikedavis** [comment]
Are you sure that it isn't being highlighted as a variable? It looks slightly brighter than the text at the beginning of the screenshot. The theme you're using might not make a big distinction for variable names (the `variable` key)

**Reply by tapyu** [reply]
On my screen, I am not able to distinguish it, but in fact, the issue is my theme (`nightfox`). When I change it, the variable is properly highlighted. I don't want to change the theme, though. Is it possible to set a different color to this element in the `config.toml`?

**Reply by the-mikedavis** [reply, marked as answer]
You can set a custom color for variables and reuse the rest of the theme by using inheritance: https://docs.helix-editor.com/master/themes.html#inheritance
